src/main/java/org/elasticsearch/index/translog/fs/FsTranslog.java

```diff
@@ -162,20 +205,25 @@ public void close() throws IOException {
 
     @Override
     public long currentId() {
-        FsTranslogFile current1 = this.current;
-        if (current1 == null) {
-            return -1;
+        try (ReleasableLock _ = readLock.acquire()) {
```

Comment:
Hmm.. does this really need a read lock? Are we concerned about the translog being swapped out underneath or something? Maybe you can add it to the javadoc, because otherwise in logging we could assume `currentId()` is zero-cost when it actually does have a cost.